test/common/testinfra.py

```diff
@@ -150,6 +150,32 @@
     'TEST_DIR',
 )
 
+def determine_github_base():
+    # pick a base
+    try:
+        # see where we get master from, e.g. origin
+        get_remote_command = ["git", "config", "--local", "--get", "branch.master.remote"]
+        remote = subprocess.Popen(get_remote_command, stdout=subprocess.PIPE, cwd=TEST_DIR).communicate()[0].strip()
+        # see if we have a git checkout - it can be in https or ssh format
+        formats = [
```

Comment:
Why don't we use ```[ "git", "ls-remote", "--get-url", remote ]``` instead of all of the screen scraping and regular expressions?